src/portal/src/lib/components/repository-gridview/repository-gridview.component.ts

```diff
@@ -129,14 +129,40 @@ export class RepositoryGridviewComponent implements OnChanges, OnInit {
             .subscribe(systemInfo => (this.systemInfo = systemInfo)
                 , error => this.errorHandler.error(error));
 
-        if (this.mode === "admiral") {
-            this.isCardView = true;
-        } else {
-            this.isCardView = false;
-        }
+        this.isCardView = this.mode === "admiral";
 
         this.lastFilteredRepoName = "";
         this.getHelmChartVersionPermission(this.projectId);
+        if (!this.searchSub) {
```

Comment:
Why add these

Reply:
I do n’t see how this helps the filter

Reply:
> Why add these

The old code used EventEmitter ,it would cause a bug.
subscribe to the input or keyup event directly to fix this bug